packages/backend/server/src/core/workspaces/resolvers/page.ts

```diff
@@ -45,8 +55,133 @@ class WorkspacePage implements Partial<PrismaWorkspacePage> {
   public!: boolean;
 }
 
+@InputType()
+class GrantDocUserRolesInput {
+  @Field(() => String)
+  docId!: string;
+
+  @Field(() => String)
+  workspaceId!: string;
+
+  @Field(() => DocRole)
+  role!: DocRole;
+
+  @Field(() => [String])
+  userIds!: string[];
+}
+
+@InputType()
+class PageGrantedUsersInput {
+  @Field(() => Int)
+  first!: number;
+
+  @Field(() => Int)
+  offset?: number;
+
+  @Field(() => String, { description: 'Cursor', nullable: true })
+  after?: string;
+
+  @Field(() => String, { description: 'Cursor', nullable: true })
+  before?: string;
+}
+
+@ObjectType()
+class GrantedDocUserType {
+  @Field(() => UserType)
+  user!: UserType;
+
+  @Field(() => DocRole)
+  role!: DocRole;
+}
+
+@ObjectType()
+class PageInfo {
+  @Field(() => String, { nullable: true })
+  startCursor?: string;
+
+  @Field(() => String, { nullable: true })
+  endCursor?: string;
+
+  @Field(() => Boolean)
+  hasNextPage!: boolean;
+
+  @Field(() => Boolean)
+  hasPreviousPage!: boolean;
+}
+
+@ObjectType()
+class GrantedDocUserEdge {
+  @Field(() => GrantedDocUserType)
+  user!: GrantedDocUserType;
+
+  @Field(() => String)
+  cursor!: string;
+}
+
+@ObjectType()
+class GrantedDocUsersConnection {
+  @Field(() => Int)
+  totalCount!: number;
+
+  @Field(() => [GrantedDocUserEdge])
+  edges!: GrantedDocUserEdge[];
+
+  @Field(() => PageInfo)
+  pageInfo!: PageInfo;
+}
+
+@ObjectType()
+export class RolePermissions
```

Comment:
it can be registered by manually apply `ObjectType` to class and `Field` to add fields. less hardcode